mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -174,6 +175,11 @@ public TaskExecutor(
             Executors.newFixedThreadPool(
                 Hardware.getNumberCPUCores(),
```

Comment:
Given that this pool is meant for IO-bound work, we can increase the number of threads to greater than the cores to address the concerns this diff is trying to solve.

Reply:
any concern on isolating main runtime task from IO work? I agree it won't cause trouble sharing with IO pool most of the time but it will be cleaner on logging/debugging and safer in case some unexpected IO work took over too many threads in future.

Reply:
Wouldn't adding an extra thread in the above line have the same effect as what you are doing? To be clear, I suggest we increase the number of IO threads to something like 3x the number of cores. 

My main concern was about the complexity. When someone looks at this code, they need to reason why we need three thread pools, and the obvious question would be, why not just use the IO pool for starting the task?

Reply:
From the code readability perspective, seeing IO-related work separated from the main execution thread is a pretty common practice right? I actually feel confused when I was analyzing the logs and saw some main entry operations were done on an IO executor thread and wondered if something went wrong with IO handling.

Reply:
I don't understand your point here - we already have the main thread executor, AFAICT. All the RPC interactions that change the state of the task executor do happen on this thread as expected. 

```
protected MainThreadExecutor getMainThreadExecutor() {
    return mainThreadExecutor;
}
```
You can see references to this executor in all the RPC interactions. Now, starting the task is not running on this executor because that's mainly IO work (such as starting the spring boot routine, etc...), and these do not change the state of the task executor.

Reply:
I see what you mean. In the TE right now:
* RPC service main thread which handles TE level interactions and state change (e.g. start task/cancel task).
* IO executor to handle resourceMgr connection IO, user code class loader setup, TE-level event queue dispatcher, and entry of the runtimeTask instance (e.g. create/maintain SB context).

- RuntimeTask level interactions cannot be part of the RPC service main thread as there are blocking operations. 

- In the meantime, the IO executor does handle a variety of work already and most of them IO related, but the actual runtimeTask service entry (e.g. SB context start) is not.

- Splitting the runtime executor means SB entry code will no longer be mixed with IO work in the same thread pool and logs, which will reduce the complexity to understand what happens in the runtimeTask lifecycle (avoid the pain of seeing all the interleaving logs on IO executor threads).

- Any future change introducing more IO work here or lifecycle changes or blocking local event dispatcher work in the runtime entry will not have unexpected blocking behavior sharing threads. This outweighs the overhead of one more executor in TE IMO.